src/yetibot/api/google.clj

```diff
@@ -0,0 +1,83 @@
+(ns yetibot.api.google
+  (:require
+   [taoensso.timbre :refer [warn info]]
+   [clojure.string :as s]
+   [clojure.data.json :as json]
+   [clj-http.client :as client]
+   [yetibot.core.config :refer [conf-valid? config-for-ns]]))
+
+(def config (config-for-ns))
+
+(defn configured? []
+  (let [config-keys   [:api-key :custom-search-engine-id]
+        config-values (select-keys config config-keys)]
+    (and (conf-valid? config)
+       (= (count config-values) 2))))
+
+(defonce api-url "https://www.googleapis.com/customsearch/v1?parameters")
+
+;; this has yet to be used somewhere
+(defonce keywords (atom {}))
```

Comment:
@devth config defaults sounds logical. There are other options like safe search and it's degree that I'm pretty sure people switch between, what search index number to start from, the way to sort the results, image sizes ... so not sure.